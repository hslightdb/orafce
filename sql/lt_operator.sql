-- test operator ||
select 'aaaaaa'::text || 'aaaaaa'::text;
select 'aaaaaa'::text || 1::integer;
select 'aaaaaa'::text || 2::smallint;
select 'aaaaaa'::text || 3::bigint;
select 'aaaaaa'::text || 1.4::numeric;
select 'aaaaaa'::text || 1.4::float4;
select 'aaaaaa'::text || 1.7::float8;
select 'aaaaaa'::text || '2014-07-20 23:59:59'::timestamp;
select 'aaaaaa'::text || '2014-07-20 23:59:57'::timestamptz;
select 'aaaaaa'::text || '2014-01-24'::date;
select 1::integer || 'aaaaaa'::text;
select 1::integer || 1::integer;
select 1::integer || 2::smallint;
select 1::integer || 3::bigint;
select 1::integer || 1.4::numeric;
select 1::integer || 1.4::float4;
select 1::integer || 1.7::float8;
select 1::integer || '2014-07-20 23:59:59'::timestamp;
select 1::integer || '2014-07-20 23:59:57'::timestamptz;
select 1::integer || '2014-01-24'::date;
select 2::smallint || 'aaaaaa'::text;
select 2::smallint || 1::integer;
select 2::smallint || 2::smallint;
select 2::smallint || 3::bigint;
select 2::smallint || 1.4::numeric;
select 2::smallint || 1.4::float4;
select 2::smallint || 1.7::float8;
select 2::smallint || '2014-07-20 23:59:59'::timestamp;
select 2::smallint || '2014-07-20 23:59:57'::timestamptz;
select 2::smallint || '2014-01-24'::date;
select 3::bigint || 'aaaaaa'::text;
select 3::bigint || 1::integer;
select 3::bigint || 2::smallint;
select 3::bigint || 3::bigint;
select 3::bigint || 1.4::numeric;
select 3::bigint || 1.4::float4;
select 3::bigint || 1.7::float8;
select 3::bigint || '2014-07-20 23:59:59'::timestamp;
select 3::bigint || '2014-07-20 23:59:57'::timestamptz;
select 3::bigint || '2014-01-24'::date;
select 1.4::numeric || 'aaaaaa'::text;
select 1.4::numeric || 1::integer;
select 1.4::numeric || 2::smallint;
select 1.4::numeric || 3::bigint;
select 1.4::numeric || 1.4::numeric;
select 1.4::numeric || 1.4::float4;
select 1.4::numeric || 1.7::float8;
select 1.4::numeric || '2014-07-20 23:59:59'::timestamp;
select 1.4::numeric || '2014-07-20 23:59:57'::timestamptz;
select 1.4::numeric || '2014-01-24'::date;
select 1.4::float4 || 'aaaaaa'::text;
select 1.4::float4 || 1::integer;
select 1.4::float4 || 2::smallint;
select 1.4::float4 || 3::bigint;
select 1.4::float4 || 1.4::numeric;
select 1.4::float4 || 1.4::float4;
select 1.4::float4 || 1.7::float8;
select 1.4::float4 || '2014-07-20 23:59:59'::timestamp;
select 1.4::float4 || '2014-07-20 23:59:57'::timestamptz;
select 1.4::float4 || '2014-01-24'::date;
select 1.7::float8 || 'aaaaaa'::text;
select 1.7::float8 || 1::integer;
select 1.7::float8 || 2::smallint;
select 1.7::float8 || 3::bigint;
select 1.7::float8 || 1.4::numeric;
select 1.7::float8 || 1.4::float4;
select 1.7::float8 || 1.7::float8;
select 1.7::float8 || '2014-07-20 23:59:59'::timestamp;
select 1.7::float8 || '2014-07-20 23:59:57'::timestamptz;
select 1.7::float8 || '2014-01-24'::date;
select '2014-07-20 23:59:59'::timestamp || 'aaaaaa'::text;
select '2014-07-20 23:59:59'::timestamp || 1::integer;
select '2014-07-20 23:59:59'::timestamp || 2::smallint;
select '2014-07-20 23:59:59'::timestamp || 3::bigint;
select '2014-07-20 23:59:59'::timestamp || 1.4::numeric;
select '2014-07-20 23:59:59'::timestamp || 1.4::float4;
select '2014-07-20 23:59:59'::timestamp || 1.7::float8;
select '2014-07-20 23:59:59'::timestamp || '2014-07-20 23:59:59'::timestamp;
select '2014-07-20 23:59:59'::timestamp || '2014-07-20 23:59:57'::timestamptz;
select '2014-07-20 23:59:59'::timestamp || '2014-01-24'::date;
select '2014-07-20 23:59:57'::timestamptz || 'aaaaaa'::text;
select '2014-07-20 23:59:57'::timestamptz || 1::integer;
select '2014-07-20 23:59:57'::timestamptz || 2::smallint;
select '2014-07-20 23:59:57'::timestamptz || 3::bigint;
select '2014-07-20 23:59:57'::timestamptz || 1.4::numeric;
select '2014-07-20 23:59:57'::timestamptz || 1.4::float4;
select '2014-07-20 23:59:57'::timestamptz || 1.7::float8;
select '2014-07-20 23:59:57'::timestamptz || '2014-07-20 23:59:59'::timestamp;
select '2014-07-20 23:59:57'::timestamptz || '2014-07-20 23:59:57'::timestamptz;
select '2014-07-20 23:59:57'::timestamptz || '2014-01-24'::date;
select '2014-01-24'::date || 'aaaaaa'::text;
select '2014-01-24'::date || 1::integer;
select '2014-01-24'::date || 2::smallint;
select '2014-01-24'::date || 3::bigint;
select '2014-01-24'::date || 1.4::numeric;
select '2014-01-24'::date || 1.4::float4;
select '2014-01-24'::date || 1.7::float8;
select '2014-01-24'::date || '2014-07-20 23:59:59'::timestamp;
select '2014-01-24'::date || '2014-07-20 23:59:57'::timestamptz;
select '2014-01-24'::date || '2014-01-24'::date;
