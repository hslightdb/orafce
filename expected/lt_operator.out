-- test operator ||
select 'aaaaaa'::text || 'aaaaaa'::text;
   ?column?   
--------------
 aaaaaaaaaaaa
(1 row)

select 'aaaaaa'::text || 1::integer;
 ?column? 
----------
 aaaaaa1
(1 row)

select 'aaaaaa'::text || 2::smallint;
 ?column? 
----------
 aaaaaa2
(1 row)

select 'aaaaaa'::text || 3::bigint;
 ?column? 
----------
 aaaaaa3
(1 row)

select 'aaaaaa'::text || 1.4::numeric;
 ?column?  
-----------
 aaaaaa1.4
(1 row)

select 'aaaaaa'::text || 1.4::float4;
 ?column?  
-----------
 aaaaaa1.4
(1 row)

select 'aaaaaa'::text || 1.7::float8;
 ?column?  
-----------
 aaaaaa1.7
(1 row)

select 'aaaaaa'::text || '2014-07-20 23:59:59'::timestamp;
            ?column?            
--------------------------------
 aaaaaaSun Jul 20 23:59:59 2014
(1 row)

select 'aaaaaa'::text || '2014-07-20 23:59:57'::timestamptz;
              ?column?              
------------------------------------
 aaaaaaSun Jul 20 23:59:57 2014 PDT
(1 row)

select 'aaaaaa'::text || '2014-01-24'::date;
     ?column?     
------------------
 aaaaaa01-24-2014
(1 row)

select 1::integer || 'aaaaaa'::text;
 ?column? 
----------
 1aaaaaa
(1 row)

select 1::integer || 1::integer;
 ?column? 
----------
 11
(1 row)

select 1::integer || 2::smallint;
 ?column? 
----------
 12
(1 row)

select 1::integer || 3::bigint;
 ?column? 
----------
 13
(1 row)

select 1::integer || 1.4::numeric;
 ?column? 
----------
 11.4
(1 row)

select 1::integer || 1.4::float4;
 ?column? 
----------
 11.4
(1 row)

select 1::integer || 1.7::float8;
 ?column? 
----------
 11.7
(1 row)

select 1::integer || '2014-07-20 23:59:59'::timestamp;
         ?column?          
---------------------------
 1Sun Jul 20 23:59:59 2014
(1 row)

select 1::integer || '2014-07-20 23:59:57'::timestamptz;
           ?column?            
-------------------------------
 1Sun Jul 20 23:59:57 2014 PDT
(1 row)

select 1::integer || '2014-01-24'::date;
  ?column?   
-------------
 101-24-2014
(1 row)

select 2::smallint || 'aaaaaa'::text;
 ?column? 
----------
 2aaaaaa
(1 row)

select 2::smallint || 1::integer;
 ?column? 
----------
 21
(1 row)

select 2::smallint || 2::smallint;
 ?column? 
----------
 22
(1 row)

select 2::smallint || 3::bigint;
 ?column? 
----------
 23
(1 row)

select 2::smallint || 1.4::numeric;
 ?column? 
----------
 21.4
(1 row)

select 2::smallint || 1.4::float4;
 ?column? 
----------
 21.4
(1 row)

select 2::smallint || 1.7::float8;
 ?column? 
----------
 21.7
(1 row)

select 2::smallint || '2014-07-20 23:59:59'::timestamp;
         ?column?          
---------------------------
 2Sun Jul 20 23:59:59 2014
(1 row)

select 2::smallint || '2014-07-20 23:59:57'::timestamptz;
           ?column?            
-------------------------------
 2Sun Jul 20 23:59:57 2014 PDT
(1 row)

select 2::smallint || '2014-01-24'::date;
  ?column?   
-------------
 201-24-2014
(1 row)

select 3::bigint || 'aaaaaa'::text;
 ?column? 
----------
 3aaaaaa
(1 row)

select 3::bigint || 1::integer;
 ?column? 
----------
 31
(1 row)

select 3::bigint || 2::smallint;
 ?column? 
----------
 32
(1 row)

select 3::bigint || 3::bigint;
 ?column? 
----------
 33
(1 row)

select 3::bigint || 1.4::numeric;
 ?column? 
----------
 31.4
(1 row)

select 3::bigint || 1.4::float4;
 ?column? 
----------
 31.4
(1 row)

select 3::bigint || 1.7::float8;
 ?column? 
----------
 31.7
(1 row)

select 3::bigint || '2014-07-20 23:59:59'::timestamp;
         ?column?          
---------------------------
 3Sun Jul 20 23:59:59 2014
(1 row)

select 3::bigint || '2014-07-20 23:59:57'::timestamptz;
           ?column?            
-------------------------------
 3Sun Jul 20 23:59:57 2014 PDT
(1 row)

select 3::bigint || '2014-01-24'::date;
  ?column?   
-------------
 301-24-2014
(1 row)

select 1.4::numeric || 'aaaaaa'::text;
 ?column?  
-----------
 1.4aaaaaa
(1 row)

select 1.4::numeric || 1::integer;
 ?column? 
----------
 1.41
(1 row)

select 1.4::numeric || 2::smallint;
 ?column? 
----------
 1.42
(1 row)

select 1.4::numeric || 3::bigint;
 ?column? 
----------
 1.43
(1 row)

select 1.4::numeric || 1.4::numeric;
 ?column? 
----------
 1.41.4
(1 row)

select 1.4::numeric || 1.4::float4;
 ?column? 
----------
 1.41.4
(1 row)

select 1.4::numeric || 1.7::float8;
 ?column? 
----------
 1.41.7
(1 row)

select 1.4::numeric || '2014-07-20 23:59:59'::timestamp;
          ?column?           
-----------------------------
 1.4Sun Jul 20 23:59:59 2014
(1 row)

select 1.4::numeric || '2014-07-20 23:59:57'::timestamptz;
            ?column?             
---------------------------------
 1.4Sun Jul 20 23:59:57 2014 PDT
(1 row)

select 1.4::numeric || '2014-01-24'::date;
   ?column?    
---------------
 1.401-24-2014
(1 row)

select 1.4::float4 || 'aaaaaa'::text;
 ?column?  
-----------
 1.4aaaaaa
(1 row)

select 1.4::float4 || 1::integer;
 ?column? 
----------
 1.41
(1 row)

select 1.4::float4 || 2::smallint;
 ?column? 
----------
 1.42
(1 row)

select 1.4::float4 || 3::bigint;
 ?column? 
----------
 1.43
(1 row)

select 1.4::float4 || 1.4::numeric;
 ?column? 
----------
 1.41.4
(1 row)

select 1.4::float4 || 1.4::float4;
 ?column? 
----------
 1.41.4
(1 row)

select 1.4::float4 || 1.7::float8;
 ?column? 
----------
 1.41.7
(1 row)

select 1.4::float4 || '2014-07-20 23:59:59'::timestamp;
          ?column?           
-----------------------------
 1.4Sun Jul 20 23:59:59 2014
(1 row)

select 1.4::float4 || '2014-07-20 23:59:57'::timestamptz;
            ?column?             
---------------------------------
 1.4Sun Jul 20 23:59:57 2014 PDT
(1 row)

select 1.4::float4 || '2014-01-24'::date;
   ?column?    
---------------
 1.401-24-2014
(1 row)

select 1.7::float8 || 'aaaaaa'::text;
 ?column?  
-----------
 1.7aaaaaa
(1 row)

select 1.7::float8 || 1::integer;
 ?column? 
----------
 1.71
(1 row)

select 1.7::float8 || 2::smallint;
 ?column? 
----------
 1.72
(1 row)

select 1.7::float8 || 3::bigint;
 ?column? 
----------
 1.73
(1 row)

select 1.7::float8 || 1.4::numeric;
 ?column? 
----------
 1.71.4
(1 row)

select 1.7::float8 || 1.4::float4;
 ?column? 
----------
 1.71.4
(1 row)

select 1.7::float8 || 1.7::float8;
 ?column? 
----------
 1.71.7
(1 row)

select 1.7::float8 || '2014-07-20 23:59:59'::timestamp;
          ?column?           
-----------------------------
 1.7Sun Jul 20 23:59:59 2014
(1 row)

select 1.7::float8 || '2014-07-20 23:59:57'::timestamptz;
            ?column?             
---------------------------------
 1.7Sun Jul 20 23:59:57 2014 PDT
(1 row)

select 1.7::float8 || '2014-01-24'::date;
   ?column?    
---------------
 1.701-24-2014
(1 row)

select '2014-07-20 23:59:59'::timestamp || 'aaaaaa'::text;
            ?column?            
--------------------------------
 Sun Jul 20 23:59:59 2014aaaaaa
(1 row)

select '2014-07-20 23:59:59'::timestamp || 1::integer;
         ?column?          
---------------------------
 Sun Jul 20 23:59:59 20141
(1 row)

select '2014-07-20 23:59:59'::timestamp || 2::smallint;
         ?column?          
---------------------------
 Sun Jul 20 23:59:59 20142
(1 row)

select '2014-07-20 23:59:59'::timestamp || 3::bigint;
         ?column?          
---------------------------
 Sun Jul 20 23:59:59 20143
(1 row)

select '2014-07-20 23:59:59'::timestamp || 1.4::numeric;
          ?column?           
-----------------------------
 Sun Jul 20 23:59:59 20141.4
(1 row)

select '2014-07-20 23:59:59'::timestamp || 1.4::float4;
          ?column?           
-----------------------------
 Sun Jul 20 23:59:59 20141.4
(1 row)

select '2014-07-20 23:59:59'::timestamp || 1.7::float8;
          ?column?           
-----------------------------
 Sun Jul 20 23:59:59 20141.7
(1 row)

select '2014-07-20 23:59:59'::timestamp || '2014-07-20 23:59:59'::timestamp;
                     ?column?                     
--------------------------------------------------
 Sun Jul 20 23:59:59 2014Sun Jul 20 23:59:59 2014
(1 row)

select '2014-07-20 23:59:59'::timestamp || '2014-07-20 23:59:57'::timestamptz;
                       ?column?                       
------------------------------------------------------
 Sun Jul 20 23:59:59 2014Sun Jul 20 23:59:57 2014 PDT
(1 row)

select '2014-07-20 23:59:59'::timestamp || '2014-01-24'::date;
              ?column?              
------------------------------------
 Sun Jul 20 23:59:59 201401-24-2014
(1 row)

select '2014-07-20 23:59:57'::timestamptz || 'aaaaaa'::text;
              ?column?              
------------------------------------
 Sun Jul 20 23:59:57 2014 PDTaaaaaa
(1 row)

select '2014-07-20 23:59:57'::timestamptz || 1::integer;
           ?column?            
-------------------------------
 Sun Jul 20 23:59:57 2014 PDT1
(1 row)

select '2014-07-20 23:59:57'::timestamptz || 2::smallint;
           ?column?            
-------------------------------
 Sun Jul 20 23:59:57 2014 PDT2
(1 row)

select '2014-07-20 23:59:57'::timestamptz || 3::bigint;
           ?column?            
-------------------------------
 Sun Jul 20 23:59:57 2014 PDT3
(1 row)

select '2014-07-20 23:59:57'::timestamptz || 1.4::numeric;
            ?column?             
---------------------------------
 Sun Jul 20 23:59:57 2014 PDT1.4
(1 row)

select '2014-07-20 23:59:57'::timestamptz || 1.4::float4;
            ?column?             
---------------------------------
 Sun Jul 20 23:59:57 2014 PDT1.4
(1 row)

select '2014-07-20 23:59:57'::timestamptz || 1.7::float8;
            ?column?             
---------------------------------
 Sun Jul 20 23:59:57 2014 PDT1.7
(1 row)

select '2014-07-20 23:59:57'::timestamptz || '2014-07-20 23:59:59'::timestamp;
                       ?column?                       
------------------------------------------------------
 Sun Jul 20 23:59:57 2014 PDTSun Jul 20 23:59:59 2014
(1 row)

select '2014-07-20 23:59:57'::timestamptz || '2014-07-20 23:59:57'::timestamptz;
                         ?column?                         
----------------------------------------------------------
 Sun Jul 20 23:59:57 2014 PDTSun Jul 20 23:59:57 2014 PDT
(1 row)

select '2014-07-20 23:59:57'::timestamptz || '2014-01-24'::date;
                ?column?                
----------------------------------------
 Sun Jul 20 23:59:57 2014 PDT01-24-2014
(1 row)

select '2014-01-24'::date || 'aaaaaa'::text;
     ?column?     
------------------
 01-24-2014aaaaaa
(1 row)

select '2014-01-24'::date || 1::integer;
  ?column?   
-------------
 01-24-20141
(1 row)

select '2014-01-24'::date || 2::smallint;
  ?column?   
-------------
 01-24-20142
(1 row)

select '2014-01-24'::date || 3::bigint;
  ?column?   
-------------
 01-24-20143
(1 row)

select '2014-01-24'::date || 1.4::numeric;
   ?column?    
---------------
 01-24-20141.4
(1 row)

select '2014-01-24'::date || 1.4::float4;
   ?column?    
---------------
 01-24-20141.4
(1 row)

select '2014-01-24'::date || 1.7::float8;
   ?column?    
---------------
 01-24-20141.7
(1 row)

select '2014-01-24'::date || '2014-07-20 23:59:59'::timestamp;
              ?column?              
------------------------------------
 01-24-2014Sun Jul 20 23:59:59 2014
(1 row)

select '2014-01-24'::date || '2014-07-20 23:59:57'::timestamptz;
                ?column?                
----------------------------------------
 01-24-2014Sun Jul 20 23:59:57 2014 PDT
(1 row)

select '2014-01-24'::date || '2014-01-24'::date;
       ?column?       
----------------------
 01-24-201401-24-2014
(1 row)

